protostar/cheatable_starknet/controllers/contracts.py

```diff
@@ -117,10 +151,50 @@ async def _transform_calldata_to_cairo_data(
                 ) from dt_exc
         return calldata or []
 
+    async def declare_sierra_contract(
+        self,
+        contract_class: ContractClass,
+        compiled_class: CompiledClass,
+        compiled_class_hash: int,
+    ) -> DeclaredSierraClass:
+        """
+        Declare a sierra contract.
+
+        @param contract_class: sierra compiled contract to be declared
+        @param compiled_class_hash: compiled class hash of the casm compiled contract.
+        @param compiled_class: casm compiled contract to be declared
+        @return: DeclaredSierraClass instance.
+        """
+        starknet_config = StarknetGeneralConfig()
+        tx = NonValidatedInternalDeclare.create(
+            contract_class=contract_class,
+            compiled_class_hash=compiled_class_hash,
+            chain_id=starknet_config.chain_id.value,
+            sender_address=DEFAULT_DECLARE_SENDER_ADDRESS,
+            max_fee=0,
+            version=2,
+            signature=[],
+            nonce=0,
+        )
+
+        with self.cheatable_state.copy_and_apply() as state_copy:
+            await tx.apply_state_updates(
+                state=state_copy, general_config=starknet_config
+            )
+
+        abi = contract_class.abi
+        # TODO figure out how it works
+        # self._add_event_abi_to_state(abi)
```

Comment:
Abis of cairo1 contract's are plain strings which is incompatible with abis we use here. Should the abi related logic be updated (and should it be done in this PR) so it's possible to to `_add_event_abi_to_state`?

Reply:
Should I leave it as todo? I think it's a okay indicator of this feature missing in case someone touches this code.

Reply:
It is ok leaving a TODO but please also create and issue

Reply:
#1663